client/src/main/scala/com.gu.contentapi.client/parser/JsonParser.scala

```diff
@@ -108,37 +109,52 @@ object Helper {
       ChangeRecord(d, user)
     }
   }
-}
-
-import Helper._
 
-object AtomQuizSerializer extends CustomSerializer[Atom](format => (
-  {
-    case rawAtom: JObject =>
+  def fixAtomsFields: PartialFunction[JField, JField] = {
+    case JField("date", JString(s)) => JField("date", JLong(new DateTime(s).getMillis))
+  }
 
-      val atomId = (rawAtom \ "id").extract[String]
-      val labels = (rawAtom \ "labels").extract[Seq[String]]
-      val defaultHtml = (rawAtom \ "defaultHtml").extract[String]
+  def createAtom(atom: JObject, atomType: AtomType, atomData: AtomData): Atoms = {
+    val atomId = (atom \ "id").extract[String]
+    val labels = (atom \ "labels").extract[Seq[String]]
+    val defaultHtml = (atom \ "defaultHtml").extract[String]
 
-      val lastModified = createChangeRecord(rawAtom, "lastModifiedDate", "lastModifiedBy")
-      val created = createChangeRecord(rawAtom, "createdDate", "createdBy")
-      val published = createChangeRecord(rawAtom, "publishDate", "publishBy")
+    val lastModified = createChangeRecord(atom, "lastModifiedDate", "lastModifiedBy")
+    val created = createChangeRecord(atom, "createdDate", "createdBy")
+    val published = createChangeRecord(atom, "publishDate", "publishBy")
 
-      val revision = (rawAtom \ "revision").extract[Long]
-      val change = ContentChangeDetails(lastModified, created, published, revision)
+    val revision = (atom \ "revision").extract[Long]
+    val change = ContentChangeDetails(lastModified, created, published, revision)
 
-      val atomData = AtomData.Quiz((rawAtom \ "data").extract[QuizAtom])
-      val atomType = AtomType.Quiz
+    Atoms(Some(Atom(atomId, atomType, labels, defaultHtml, atomData, change)))
+  }
+}
 
-      Atom(atomId, atomType, labels, defaultHtml, atomData, change)
+import Helper._
 
+object AtomSerializer extends CustomSerializer[Atoms](format => (
+  {
+    case rawAtom: JObject =>
+      if((rawAtom \ "quiz") != JNothing) {
+        val quizAtom: JObject = (rawAtom \ "quiz").extract[JObject]
+        val atomData = AtomData.Quiz((quizAtom \ "data").extract[QuizAtom])
+        createAtom(quizAtom, AtomType.Quiz, atomData)
+      }
+      else if((rawAtom \ "viewpoints") != JNothing) {
+        val viewpointAtom: JObject = (rawAtom \ "viewpoints").extract[JObject]
```

Comment:
sorry I was wrong on this, viewpoints returned currently a viewpoints atom object.